src/Idris/Syntax.idr

```diff
@@ -574,12 +574,16 @@ data EditCmd : Type where
      TypeAt : Int -> Int -> Name -> EditCmd
      CaseSplit : Bool -> Int -> Int -> Name -> EditCmd
      AddClause : Bool -> Int -> Name -> EditCmd
-     ExprSearch : Bool -> Int -> Name -> List Name -> EditCmd
+     ExprSearch : Bool -> Int -> String -> List Name -> EditCmd
      ExprSearchNext : EditCmd
      GenerateDef : Bool -> Int -> Name -> Nat -> EditCmd
      GenerateDefNext : EditCmd
-     MakeLemma : Bool -> Int -> Name -> EditCmd
-     MakeCase : Bool -> Int -> Name -> EditCmd
+     MakeLemma : Bool -> Int ->
+                 String -> -- hole name, without the question mark
+                 EditCmd
+     MakeCase : Bool -> Int ->
+                String -> --hole name, without the question mark
```

Comment:
It feels like we lost something in the way of self documenting types with the changes from Name to String in a number of places. What was the motivation for this change?